src/UI/class-recommended-widget.php

```diff
@@ -259,26 +290,30 @@ class="tiny-text" aria-labelledby="<?php echo esc_attr( $this->get_field_id( 'pu
 			<label for="<?php echo esc_attr( $this->get_field_id( 'personalize_results' ) ); ?>"><?php esc_html_e( 'Personalize recommended results', 'wp-parsely' ); ?></label>
 		</p>
 		<?php
+
+		return '';
 	}
 
 	/**
 	 * This is the update function.
 	 *
-	 * @param array $new_instance The new values for the db.
-	 * @param array $old_instance Values saved to the db.
-	 * @return array
+	 * @param WidgetSettings $new_instance The new values for the db.
+	 * @param WidgetSettings $old_instance Values saved to the db.
+	 *
+	 * @return WidgetSettings
 	 */
-	public function update( $new_instance, $old_instance ): array {
+	public function update( $new_instance, $old_instance ) /* @phpstan-ignore-line */ {
```

Comment:
Same as above i.e. enforcing types instead of using WP default function signature.